locale/en/docs/guides/backpressuring-in-streams.md

```diff
@@ -0,0 +1,264 @@
+---
+title: Backpressuring in Streams
+layout: docs.hbs
+---
+
+# Backpressuring in Streams
+ 
+The purpose of this guide will describe what backpressure is and the 
+problems in Streams that it solves. The first part of this guide will cover 
+these questions. The second part of the guide will introduce suggested best 
+practices to ensure your application's code is safe and optimized.
+
+We assume a little familiarity with the general definition of 
+[`backpressure`][], [`Buffer`][] and [`EventEmitters`][] in Node.js, as well as 
+some experience with [`Streams`][]. If you haven't read through those docs, 
+it's not a bad idea to take a look at the API documentation first, as it will 
+help expand your understanding while reading this guide and for following along 
+with examples.
+
+## The Problem with Streams
+
+In a computer system, data is transferred from one process to another through 
+pipes and signals. In Node.js, we find a similar mechanism and called 
+[`Streams`][]. Streams are great! They do so much for Node.js and almost every
+part of the internal codebase utilizes the `Streams` class. As a developer, you 
+are more than encouraged to use them too!
+
+```javascript
+const readline = require('readline');
+
+// process.stdin and process.stdout are both instances of Streams
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+rl.question('Why should you use streams? ', (answer) => {
+  
+  console.log(`Maybe it's ${answer}, maybe it's because they are awesome! :)`);
+  
+  rl.close();
+});
+```
+
+
+There are different functions to transfer data from one process to another. In 
+Node.js, there is an internal built-in function called [`.pipe()`][]. There are
+[other packages][] out there you can use too! Ultimately though, at the basic 
+level of this process, we have two separate components: the _source_ of the 
+data and the _consumer_.
+
+In Node.js the source is a [`Readable`][] stream and the consumer is the 
+[`Writable`][] stream (both of these may be interchanged with a [`Duplex`][] 
+stream, but that is out-of-scope for this guide).
+
+## Too Much Data, Too Quickly
+
+There are instance where a [`Readable`][] stream might give data to the
+[`Writable`][] much too quickly --- much more than the consumer can handle! 
+
+When that occurs, the consumer will begin to queue all the chunks of data for 
+later consumption. The write queue will get longer and longer, and because of 
+this more data must be kept in memory until the entire process has completed.
+
+```javascript
+  var fs = require('fs');
+
+  var inputFile  = fs.createReadStream('REALLY_BIG_FILE.x');
+  var outputFile = fs.createWriteStream('REALLY_BIG_FILE_DEST.x');
+
+  // Secretly the stream is saying: "whoa, whoa! hang on, this is way too much!"
```

Comment:
Writing to the disk is slower than reading from it. It might not be clear for a reader though, so I'm 👍  in adding a zlib compression before writing down.